File: src/main/java/com/example/testing/app/GreetingController.java
package com.example.testing.app;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class GreetingController {
    @GetMapping("/hi")
    public String dummy() {
        return "{\"message\":\"Hi!\"}";
    }
}

File: src/main/kotlin/com/example/testing/GreetingController.kt
package com.example.testing

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@Controller
//@RestController
class GreetingController {
    @GetMapping("/hello")
    fun hello(model: org.springframework.ui.Model):HelloResponse {
        return HelloResponse("hello")
    }
}